pkg/edgexsdk/edgexsdk.go

```diff
@@ -205,26 +208,44 @@ func (sdk *AppFunctionsSDK) initializeRegistry() error {
 			return fmt.Errorf("could not register service with Registry: %v", err)
 		}
 
-		rawConfig, err := sdk.registryClient.GetConfiguration(configuration)
+		hasConfig, err := sdk.registryClient.HasConfiguration()
 		if err != nil {
-			return fmt.Errorf("could not get configuration from Registry: %v", err)
+			return fmt.Errorf("could not determine if registry has configuration: %v", err)
 		}
 
-		actual, ok := rawConfig.(*common.ConfigurationStruct)
-		if !ok {
-			return fmt.Errorf("configuration from Registry failed type check")
-		}
+		if hasConfig {
+			rawConfig, err := sdk.registryClient.GetConfiguration(configuration)
+			if err != nil {
+				return fmt.Errorf("could not get configuration from Registry: %v", err)
+			}
+
+			actual, ok := rawConfig.(*common.ConfigurationStruct)
```

Comment:
Recommend you also check to see if the struct is empty or not.
https://github.com/edgexfoundry/edgex-go/blob/aac6afd9dc210ebc10da08c4cdc86a425f9683b4/internal/core/data/init.go#L196

Reply:
Should be updated now